rust/src/aggregator/rpc.rs

```diff
@@ -28,7 +28,43 @@ mod inner {
     }
 }
 
-pub use inner::{Rpc, RpcClient as Client};
+pub use inner::Rpc;
+
+#[cfg(not(test))]
+pub use inner::RpcClient as Client;
+
+#[cfg(test)]
+mod mocks {
+    pub use super::*;
+
+    use futures::future;
+    use mockall::mock;
+    use std::io;
+    use tarpc::{client::Config, context::Context, rpc::Transport};
+
+    mock! {
+        pub NewClient {
+            fn spawn(self) -> io::Result<MockClient>;
+        }
+    }
+
+    mock! {
+        pub Client {
```

Comment:
for readability, i would suggest to name the mocked object exactly the same as the real ones and assign an alias later by `use`. i.e.
```
mock! { pub RpcClient {...} }

#[cfg(test)]
pub use mocks::MockRpcClient as Client;
```
this makes it clearer what is to be mocked here. same for `NewRpcClient` instead of `RpcClient`.

Reply:
Good point. I'll do that.